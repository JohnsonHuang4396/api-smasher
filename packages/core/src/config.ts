import { normalize } from 'pathe'

export const DEFAULT_OUTPUT_DIR = normalize('/src/api')

export const DEFAULT_API_TEMPLATE_PATH = normalize('./generator/template/web-fetch/index.ejs')
