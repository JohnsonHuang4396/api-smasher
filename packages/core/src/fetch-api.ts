import type { SwaggerConfig, SwaggerDocument, SwaggerEndpoint, SwaggerResponse } from './types'

export async function fetchSwaggerDocs(config: SwaggerConfig): Promise<SwaggerResponse[]> {
  const endpoints = normalizeConfig(config)
  const responses: SwaggerResponse[] = []

  for (const endpoint of endpoints) {
    try {
      const response = await fetchEndpoint(endpoint)
      responses.push(response)
    }
    catch (error) {
      if (error instanceof Error) {
        throw new TypeError(`❌ Failed to fetch swagger doc from ${endpoint.remote}: ${error.message}`)
      }
      throw error
    }
  }

  return responses
}

function normalizeConfig(config: SwaggerConfig): SwaggerEndpoint[] {
  if (typeof config === 'string') {
    return [{ remote: config }]
  }

  if (Array.isArray(config)) {
    return config.map(item =>
      typeof item === 'string' ? { remote: item } : item
    )
  }

  return [config]
}

async function fetchEndpoint(endpoint: SwaggerEndpoint): Promise<SwaggerResponse> {
  const { remote, header, response: responseHandler } = endpoint

  const fetchOptions: RequestInit = {
    headers: header
  }

  const fetchResponse = await fetch(remote, fetchOptions)

  if (!fetchResponse.ok) {
    throw new Error(`❌ HTTP error! status: ${fetchResponse.status}`)
  }

  const data = (await fetchResponse.json()) as SwaggerDocument

  if (responseHandler) {
    return await responseHandler(data)
  }

  return {
    remote,
    name: extractNameFromRemote(remote),
    paths: data.paths,
    components: data.components
  }
}

function extractNameFromRemote(remote: string): string {
  try {
    const url = new URL(remote)
    return url.hostname.split('.')[0]
  }
  catch {
    return remote.split('/').pop() || remote
  }
}
