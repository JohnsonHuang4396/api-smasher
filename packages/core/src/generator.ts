import type { SwaggerResponse } from './types'
import { appendFileSync, existsSync, mkdirSync, writeFileSync } from 'node:fs'
import { cwd } from 'node:process'
import { normalize, resolve } from 'pathe'

interface GeneratorOptions {
  outputDir: string
  pathDetails: {
    method: string
    path: string
    operationId?: string
  }[]
  responses: SwaggerResponse[]
}

function generateFiles(outputDir: string, path: string, content: string): void {
  if (existsSync(path)) {
    appendFileSync(path, `\n${content}`)
  }
  else {
    mkdirSync(outputDir, { recursive: true })
    writeFileSync(path, content)
  }
}

export function generateApiFiles({ outputDir, pathDetails, responses }: GeneratorOptions): void {
  const basePath = normalize(resolve(cwd(), outputDir))
  const apiPath = normalize(resolve(basePath, 'Api.ts'))
  const modelPath = normalize(resolve(basePath, 'Model.ts'))

  // 生成 API 文件
  const apiContent = generateApiContent(pathDetails)
  generateFiles(outputDir, apiPath, apiContent)

  // 生成 Model 文件
  const modelContent = generateModelContent(responses)
  generateFiles(outputDir, modelPath, modelContent)
}

function generateApiContent(pathDetails: GeneratorOptions['pathDetails']): string {
  return pathDetails.map((detail) => {
    const functionName = detail.operationId || getFunctionName(detail.path)
    const method = detail.method.toLowerCase()

    return `
export async function ${functionName}(params: any) {
  return await fetch('${detail.path}', {
    method: '${method}',
    headers: {
      'Content-Type': 'application/json'
    },
    body: ${method !== 'get' ? 'JSON.stringify(params)' : 'undefined'}
  }).then(res => res.json())
}
`
  }).join('\n')
}

function generateModelContent(responses: SwaggerResponse[]): string {
  // 从responses中提取所有需要的类型定义
  const schemas = responses.flatMap(response =>
    Object.entries(response.components?.schemas || {})
  )

  return schemas.map(([name, schema]) => {
    return `
export interface ${name} {
${generateTypeProperties(schema)}
}
`
  }).join('\n')
}

function getFunctionName(path: string): string {
  const parts = path.split('/')
  return parts[parts.length - 1] || 'api'
}

function getTypeFromSchema(prop: any): string {
  if (prop.$ref) {
    return prop.$ref.split('/').pop()
  }

  switch (prop.type) {
    case 'string':
      return 'string'
    case 'integer':
    case 'number':
      return 'number'
    case 'boolean':
      return 'boolean'
    case 'array':
      return `${getTypeFromSchema(prop.items)}[]`
    case 'object':
      return 'Record<string, any>'
    default:
      return 'any'
  }
}

function getPropertyDefinition(name: string, type: string, optional: string): string {
  let extra = ''
  if (name.toLowerCase().endsWith('id')) {
    extra = `_${name}${optional}: ${type};`
  }
  return `${name}${optional}: ${type};\n${extra}`
}

function generateTypeProperties(schema: any): string {
  if (!schema.properties)
    return ''

  return Object.entries(schema.properties)
    .map(([name, prop]: [string, any]) => {
      const type = getTypeFromSchema(prop)
      const optional = schema.required?.includes(name) ? '' : '?'
      const description = prop.description || name
      return `
  /**
  * ${description}
  */
  ${getPropertyDefinition(name, type, optional)}`
    })
    .join('\n')
}
