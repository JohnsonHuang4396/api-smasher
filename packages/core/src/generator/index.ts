import type { SwaggerPathDefinition, SwaggerResponse } from '../types'
import type { ApiGeneratorOptions } from './api-generator'
import { appendFileSync, existsSync, mkdirSync, writeFileSync } from 'node:fs'
import { cwd } from 'node:process'
import { normalize, resolve } from 'pathe'
import { generateApiContent } from './api-generator'
import { generateModelContent } from './model-generator'

export interface GeneratorOptions {
  outputDir: string
  pathDetails: {
    method: string
    path: string
    raw: SwaggerPathDefinition
    operationId?: string
    summary?: string
  }[]
  responses: SwaggerResponse[]
  generateConfig: ApiGeneratorOptions
}

function generateFiles(outputDir: string, path: string, content: string): void {
  if (existsSync(path)) {
    appendFileSync(path, `\n\n${content}`)
  }
  else {
    mkdirSync(outputDir, { recursive: true })
    writeFileSync(path, content)
  }
}

export async function generateApiFiles({
  outputDir,
  pathDetails,
  responses,
  generateConfig
}: GeneratorOptions): Promise<void> {
  const basePath = normalize(resolve(cwd(), outputDir))
  const apiPath = normalize(resolve(basePath, 'Api.ts'))
  const modelPath = normalize(resolve(basePath, 'Model.ts'))

  const apiContent = await generateApiContent(pathDetails, generateConfig)
  generateFiles(outputDir, apiPath, apiContent)

  const modelContent = generateModelContent(responses)
  generateFiles(outputDir, modelPath, modelContent)
}
