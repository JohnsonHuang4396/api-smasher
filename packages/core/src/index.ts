export * from './action'
export * from './fetch-api'
export * from './generator'
export * from './prompt'
export * from './types'
