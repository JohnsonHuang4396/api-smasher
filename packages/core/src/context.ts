import type { ActionOptions, Template } from './action'
import type { SwaggerResponse } from './types'
import { DEFAULT_OUTPUT_DIR, DEFAULT_WEB_REQUEST_TEMPLATE_PATH } from './config'

class GeneratorContext {
  private _templates: Template[] = [
    {
      name: 'default-web-fetch',
      path: DEFAULT_WEB_REQUEST_TEMPLATE_PATH
    }
  ]

  private _responses: SwaggerResponse[] = []

  constructor(
    public readonly options: ActionOptions,
    responses?: SwaggerResponse[]
  ) {
    if (Array.isArray(this.options.generateConfig?.api?.template)) {
      this.templates.push(...this.options.generateConfig.api.template)
    }
    if (responses) {
      this.responses = responses
    }
  }

  get outputDir(): string {
    return this.options.outputDir ?? DEFAULT_OUTPUT_DIR
  }

  get templates(): Template[] {
    return this._templates
  }

  get responses(): SwaggerResponse[] {
    return this._responses
  }

  set responses(responses: SwaggerResponse[]) {
    this._responses = responses
  }

  get selectedPaths(): string[] {
    return this.options.selectedPaths ?? []
  }

  get generateConfig(): ActionOptions['generateConfig'] {
    return this.options.generateConfig ?? {}
  }
}

let context: GeneratorContext | null = null

export function createGeneratorContext(options: ActionOptions, responses?: SwaggerResponse[]): GeneratorContext {
  context = new GeneratorContext(options, responses)
  return context
}

export function getGeneratorContext(): GeneratorContext {
  if (!context) {
    throw new Error('Generator context not found')
  }
  return context
}
