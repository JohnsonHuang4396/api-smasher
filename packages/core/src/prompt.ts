import type { ActionOptions } from './action'
import type { GeneratorOptions } from './generator'
import type { SwaggerResponse } from './types'
import { existsSync, mkdirSync } from 'node:fs'
import process from 'node:process'
import { confirm, intro, outro, select, spinner, text } from '@clack/prompts'
import { normalize, resolve } from 'pathe'
import { DEFAULT_OUTPUT_DIR } from './config'
import { getGeneratorContext } from './context'

export interface PromptConfig {
  responses: SwaggerResponse[]
  actionOptions: PromptActionOptions
}

export interface PromptActionOptions extends Omit<
  ActionOptions,
'selectedPaths' | 'autoRun'
> { }

export async function promptUserInteraction({
  responses,
  actionOptions
}: PromptConfig): Promise<{
    selectedPaths: string[]
    outputDir: string
    pathDetails: GeneratorOptions['pathDetails']
    generateConfig: ActionOptions['generateConfig']
  }> {
  try {
    intro('API Smasher - Swagger to TypeScript converter')

    // 1. 获取所有可用的API路径
    const availablePaths = responses.flatMap(response =>
      Object.entries(response.paths || {}).flatMap(([path, methods]) =>
        Object.entries(methods as Record<string, any>).map(([method, info]) => ({
          method: method.toUpperCase(),
          path,
          summary: info.summary,
          operationId: info.operationId,
          raw: info
        }))
      )
    )

    // 2. 询问用户需要创建哪些接口
    const pathInput = await text({
      message: 'Enter the API paths to generate (comma separated)',
      placeholder: '/api/test,/api/create',
      validate(value) {
        if (!value)
          return 'Please enter at least one API path'
      }
    }) as string

    if (pathInput === null) {
      outro('Operation cancelled')
      process.exit(0)
    }

    // 3. 解析用户输入的路径
    const s = spinner()
    s.start('Validating API paths')

    const selectedPaths = pathInput.split(',').map(p => p.trim())
    const validPaths = selectedPaths.filter((path) => {
      return availablePaths.some(p => p.path === path)
    })

    if (validPaths.length === 0) {
      s.stop('No valid API paths found')
      outro('Please check your input and try again')
      process.exit(1)
    }

    if (validPaths.length !== selectedPaths.length) {
      const invalidPaths = selectedPaths.filter(p => !validPaths.includes(p))
      s.stop(`Warning: Some paths are invalid: ${invalidPaths.join(', ')}`)
    }
    else {
      s.stop('All paths are valid')
    }

    // 4. 询问输出目录
    const outputDir = await text({
      message: 'Where would you like to create the API files?',
      placeholder: actionOptions?.outputDir || DEFAULT_OUTPUT_DIR,
      initialValue: actionOptions?.outputDir || DEFAULT_OUTPUT_DIR,
      validate(value) {
        if (!value)
          return 'Please enter a directory path'
      }
    }) as string

    if (outputDir === null) {
      outro('Operation cancelled')
      process.exit(0)
    }

    // 确保目录存在
    const fullPath = normalize(resolve(process.cwd(), outputDir))
    if (!existsSync(fullPath)) {
      const shouldCreate = await confirm({
        message: `Directory ${outputDir} does not exist. Create it?`
      })

      if (shouldCreate) {
        try {
          mkdirSync(fullPath, { recursive: true })
        }
        catch (error) {
          outro(`❌ Failed to create directory: ${error instanceof Error ? error.message : String(error)}`)
          process.exit(1)
        }
      }
      else {
        outro('❌ Operation cancelled')
        process.exit(0)
      }
    }

    let apiTemplatePath = ''
    apiTemplatePath = await select({
      message: 'Do you want to use a custom API template?',
      options: [
        ...(getGeneratorContext().templates.map((template) => {
          if (typeof template === 'string') {
            return {
              value: template,
              label: template
            }
          }
          return {
            value: template.name,
            label: template.name,
            hint: template.path
          }
        }) || []),
        { value: 'custom', label: 'Custom' }
      ]
    }) as string

    if (apiTemplatePath === 'custom') {
      apiTemplatePath = await text({
        message: 'Enter the path to your custom API template',
        placeholder: '~/path/to/template.ejs',
        validate(value) {
          if (!value)
            return 'Please enter a valid path'
        }
      }) as string
    }

    // 5. 返回用户选择的信息
    return {
      selectedPaths: validPaths,
      outputDir: fullPath,
      pathDetails: availablePaths.filter(info =>
        validPaths.includes(info.path)
      ),
      generateConfig: {
        ...actionOptions.generateConfig,
        api: {
          template: [apiTemplatePath]
        }
      }
    }
  }
  catch (error) {
    outro(`❌ Failed to prompt user interaction: ${error instanceof Error ? error.message : String(error)}`)
    process.exit(1)
  }
}
