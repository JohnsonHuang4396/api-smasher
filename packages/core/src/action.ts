import type { GeneratorOptions } from './generator'
import type { SwaggerConfig, SwaggerResponse } from './types'
import { select } from '@clack/prompts'
import { DEFAULT_OUTPUT_DIR } from './config'
import { fetchSwaggerDocs } from './fetch-api'
import { generateApiFiles } from './generator'
import { promptUserInteraction } from './prompt'

export interface ActionOptions {
  /**
   * Swagger API 文档地址配置
   */
  config: SwaggerConfig
  /**
   * 是否自动执行（不需要用户交互）
   */
  autoRun?: boolean
  /**
   * 自动执行时的输出目录
   */
  outputDir?: string
  /**
   * 自动执行时要生成的API路径列表
   */
  selectedPaths?: string[]
  /**
   * 生成配置
   */
  generateConfig?: {
    /**
     * 是否生成模型文件
     */
    model?: boolean
    /**
     * API 生成配置
     */
    api?: {
      /**
       * 模板路径
       */
      template?: string[] | { name: string, path: string }[]
    }
  }
}

export async function generateApi({
  config,
  autoRun = false,
  outputDir = DEFAULT_OUTPUT_DIR,
  selectedPaths = [],
  generateConfig = {
    model: true,
    api: {}
  }
}: ActionOptions): Promise<void> {
  try {
    // 1. 获取 Swagger 文档
    const responses = await fetchSwaggerDocs(config)

    // 2. 如果不是自动运行，则进行用户交互
    if (!autoRun) {
      const result = await promptUserInteraction({
        responses,
        actionOptions: {
          config,
          outputDir,
          generateConfig
        }
      })
      return await processApiGeneration(responses, result)
    }

    // 3. 自动运行模式
    const availablePaths = extractAvailablePaths(responses)

    // 验证选择的路径是否有效
    const validPaths = validateSelectedPaths(selectedPaths, availablePaths)
    if (validPaths.length === 0)
      throw new Error('❌ No valid API paths selected')

    // 4. 生成文件
    return await processApiGeneration(responses, {
      outputDir,
      selectedPaths: validPaths,
      pathDetails: availablePaths.filter(info =>
        validPaths.includes(info.path)
      ),
      generateConfig
    })
  }
  catch (error) {
    if (error instanceof Error)
      throw new Error(`❌ Failed to generate API: ${error.message}`)
    throw error
  }
}

interface ProcessResult {
  outputDir: string
  selectedPaths: string[]
  pathDetails: GeneratorOptions['pathDetails']
  generateConfig: ActionOptions['generateConfig']
}

async function processApiGeneration(
  responses: SwaggerResponse[],
  result: ProcessResult
): Promise<void> {
  try {
    const { outputDir, pathDetails, generateConfig } = result
    let template: string | undefined
    if (Array.isArray(generateConfig?.api?.template) && generateConfig.api.template.length > 1) {
      template = await select({
        message: 'Select the API template',
        options: generateConfig.api.template.map((template) => {
          if (typeof template === 'string') {
            return {
              value: template,
              label: template
            }
          }
          return {
            value: template.name,
            label: template.name,
            hint: template.path
          }
        })
      }) as string
    }
    await generateApiFiles({
      outputDir,
      pathDetails,
      responses,
      generateConfig: {
        ...generateConfig,
        template
      }
    })
  }
  catch (error) {
    if (error instanceof Error)
      throw new Error(`❌ Failed to generate API files: ${error.message}`)
    throw error
  }
}

function extractAvailablePaths(responses: SwaggerResponse[]): Array<ProcessResult['pathDetails'][number]> {
  return responses.flatMap(response =>
    Object.entries(response.paths || {}).flatMap(([path, methods]) =>
      Object.entries(methods as Record<string, any>).map(([method, info]) => {
        const { summary, operationId } = info
        return {
          method: method.toUpperCase(),
          path,
          summary,
          operationId,
          raw: info
        }
      })
    )
  )
}

function validateSelectedPaths(
  selectedPaths: string[],
  availablePaths: Array<{ method: string, path: string }>
): string[] {
  const validPaths = new Set(
    availablePaths.map(info => info.path)
  )

  return selectedPaths.filter(path => validPaths.has(path))
}
