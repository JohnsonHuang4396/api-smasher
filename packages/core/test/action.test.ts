import * as fs from 'node:fs'
import * as path from 'node:path'
import { beforeEach, describe, expect, it, vi } from 'vitest'
import { generateApi } from '../src/action'

const REAL_SWAGGER_URL = 'http://192.168.100.125:18083/v3/api-docs'
const TEST_OUTPUT_DIR = './test/test-output/api'

describe('generateApi', { only: true }, () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  it('should generate API files in auto-run mode and match snapshot', async () => {
    if (fs.existsSync(TEST_OUTPUT_DIR)) {
      fs.rmSync(TEST_OUTPUT_DIR, { recursive: true })
    }

    const selectedPaths = [
      '/wechatContentMsg/getWechatAssessmentByEmployee',
      '/wechatContentMsg/getWechatAssessmentByClient'
    ]

    await generateApi({
      config: REAL_SWAGGER_URL,
      autoRun: true,
      outputDir: TEST_OUTPUT_DIR,
      selectedPaths
    })

    // 验证文件是否生成
    expect(fs.existsSync(path.join(TEST_OUTPUT_DIR, 'Api.ts'))).toBe(true)
    expect(fs.existsSync(path.join(TEST_OUTPUT_DIR, 'Model.ts'))).toBe(true)

    // 读取生成的文件内容
    const apiContent = fs.readFileSync(path.join(TEST_OUTPUT_DIR, 'Api.ts'), 'utf-8')
    const modelContent = fs.readFileSync(path.join(TEST_OUTPUT_DIR, 'Model.ts'), 'utf-8')

    // 将生成的文件内容与快照比对
    expect(apiContent).toMatchSnapshot('generated-api-content')
    expect(modelContent).toMatchSnapshot('generated-model-content')
  })

  it('should handle invalid paths in auto-run mode', async () => {
    const selectedPaths = ['POST:/invalid/path']

    await expect(generateApi({
      config: REAL_SWAGGER_URL,
      autoRun: true,
      outputDir: TEST_OUTPUT_DIR,
      selectedPaths
    })).rejects.toBeTruthy()
  })

  it('should handle network errors', async () => {
    const invalidUrl = 'http://invalid-url.example.com/swagger'

    await expect(generateApi({
      config: invalidUrl,
      autoRun: true,
      outputDir: TEST_OUTPUT_DIR,
      selectedPaths: []
    })).rejects.toBeTruthy()
  })

  it('should handle file system errors', async () => {
    const invalidDir = '/invalid/directory'
    const selectedPaths = [
      'POST:/wechatContentMsg/getWechatAssessmentByEmployee'
    ]

    await expect(generateApi({
      config: REAL_SWAGGER_URL,
      autoRun: true,
      outputDir: invalidDir,
      selectedPaths
    })).rejects.toBeTruthy()
  })
})
