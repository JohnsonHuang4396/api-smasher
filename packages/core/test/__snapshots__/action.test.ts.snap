// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateApi > should generate API files in auto-run mode and match snapshot > generated-api-content 1`] = `
"export async function getWechatAssessmentByEmployee(body: WechatAssessmentQO) {
  const result = await axios.post<RIPageWechatEmployeeConversationVO>(
    "/api/wechatContentMsg/getWechatAssessmentByEmployee",
    body
  );
  return result.data.data;
}
export async function getWechatAssessmentByClient(body: WechatAssessmentQO) {
  const result = await axios.post<RIPageWechatClientConversationVO>(
    "/api/wechatContentMsg/getWechatAssessmentByClient",
    body
  );
  return result.data.data;
}"
`;

exports[`generateApi > should generate API files in auto-run mode and match snapshot > generated-model-content 1`] = `
"
export interface WechatAssessmentQO {
/**
* 当前页
*/
current?: number;

/**
* 每页的数量
*/
size?: number;

/**
* 升序字段：用','隔开
*/
ascStr?: string;

/**
* 降序字段：用','隔开
*/
descStr?: string;

/**
* 企微群id
*/
chat_id?: string;
_chat_id?: string;
/**
* 企微群名称
*/
name?: string;

/**
* crm客户id
*/
clientId?: string;
_clientId?: string;
/**
* crm客户名称
*/
clientName?: string;

/**
* 亿企赢绑定手机号
*/
loginName?: string;

/**
* 员工id
*/
userId?: string;
_userId?: string;
/**
* 员工名称
*/
userName?: number;

/**
* 开始时间
*/
startDate?: string;

/**
* 结束时间
*/
endDate?: string;

}


export interface IPageWechatEmployeeConversationVO {
/**
* size
*/
size?: number;

/**
* records
*/
records?: WechatEmployeeConversationVO[];

/**
* current
*/
current?: number;

/**
* total
*/
total?: number;

/**
* pages
*/
pages?: number;

}


export interface RIPageWechatEmployeeConversationVO {
/**
* code
*/
code?: number;

/**
* success
*/
success?: boolean;

/**
* data
*/
data?: IPageWechatEmployeeConversationVO;

/**
* msg
*/
msg?: string;

}


export interface WechatEmployeeConversationVO {
/**
* 员工姓名
*/
userName?: string;

/**
* 所属组织Id
*/
orgId?: string;
_orgId?: string;
/**
* 联系客户数
*/
contactClientNum?: number;

/**
* 活跃客户数
*/
activeClientNum?: number;

/**
* 沉默客户数
*/
inactiveClientNum?: number;

/**
* 咨询会话数
*/
clientContactNum?: number;

/**
* 回复会话数
*/
replyClientNum?: number;

/**
* 未回复会话数
*/
noReplyClientNum?: number;

/**
* 咨询回复率
*/
replyRatio?: string;

/**
* 平均会话时长
*/
avgDuration?: string;

/**
* 发送消息总数
*/
contactClientMsgNum?: number;

/**
* 主动发送消息数
*/
proactiveMsgNum?: number;

/**
* 与客户沟通确认申报税款通知次数
*/
taxConfirmNum?: number;

}


export interface IPageWechatClientConversationVO {
/**
* size
*/
size?: number;

/**
* records
*/
records?: WechatClientConversationVO[];

/**
* current
*/
current?: number;

/**
* total
*/
total?: number;

/**
* pages
*/
pages?: number;

}


export interface RIPageWechatClientConversationVO {
/**
* code
*/
code?: number;

/**
* success
*/
success?: boolean;

/**
* data
*/
data?: IPageWechatClientConversationVO;

/**
* msg
*/
msg?: string;

}


export interface WechatClientConversationVO {
/**
* 企业名称
*/
clientName?: string;

/**
* 所属组织Id
*/
orgId?: string;
_orgId?: string;
/**
* 员工发送消息数
*/
employeeMsgNum?: number;

/**
* 客户发送消息数
*/
clientMsgNum?: number;

/**
* 咨询会话数
*/
clientContactNum?: number;

/**
* 回复会话数
*/
replyClientNum?: number;

/**
* 未回复会话数
*/
noReplyClientNum?: number;

/**
* 咨询回复率
*/
replyRatio?: string;

/**
* 平均会话时长
*/
avgDuration?: string;

/**
* 关联企微群名称
*/
weChatGroupName?: string;

/**
* 是否及时建群
*/
isPromptlySetUp?: string;

/**
* 与客户沟通确认申报税款通知次数
*/
taxConfirmNum?: number;

}
"
`;
